### DISCUSSION TITLE
ndphlpr.vxd thoughts

### DISCUSSION THREAD

**Discussion by yincrash** [itsmattkc/dotnet9x#28] [General]
A good mystery is always too alluring to try and solve.

NDP stands for .NET Development Platform. HLPR stands for "helper".

I would then guess that the driver is there to give .NET access to the kernel and hardware.

No idea what the request code and response from the YT video is for, but if I were to guess, since it's on startup, it's a check to see what version of the .vxd is installed to know what features the current version has access to.

**Comment by IoIxD** [comment]
This shows up on Google now so I just want to clarify here that this was solved. It exists to fix a bug in 9x related to working with threads.

**Comment by yincrash** [comment]
See https://github.com/itsmattkc/dotnet9x/issues/14

**Comment by DimaLeon2000** [comment]
**NDP** may also stand for "**N**T **D**evelopment **P**latform".
